lib/publiccloud/provider.pm

```diff
@@ -149,8 +149,11 @@ Creates an ssh keypair in a given file path by $args{ssh_private_key_file}
 sub create_ssh_key {
     my ($self, %args) = @_;
     $args{ssh_private_key_file} //= '/root/.ssh/id_rsa';
+    # ssh user@instance awk 'NF {sub(/\r/, ""); printf "%s\\n",$0;}' ~/.ssh/id_rsa; echo
+    my $ssh_key = get_var('PUBLIC_CLOUD_EXISTING_SSH_KEY');
+    assert_script_run('SSH_DIR=`dirname ' . $args{ssh_private_key_file} . '`; mkdir -p $SSH_DIR');
+    assert_script_run("echo -e \"$ssh_key\" | tee $args{ssh_private_key_file}; chmod 0700 $args{ssh_private_key_file}") if ($ssh_key);
     if (script_run('test -f ' . $args{ssh_private_key_file}) != 0) {
-        assert_script_run('SSH_DIR=`dirname ' . $args{ssh_private_key_file} . '`; mkdir -p $SSH_DIR');
```

Comment:
This whole thing is a bit confusing now... Would it make sense to do something like
```
my $ssh_key = get_var('PUBLIC_CLOUD_EXISTING_SSH_KEY');
if ($ssh_key) { 
  # do something
} elsif (script_run('test -f ' . $args{ssh_private_key_file}) != 0) {
  assert_script_run('SSH_DIR=`dirname ' . $args{ssh_private_key_file} . '`; mkdir -p $SSH_DIR');
  assert_script_run('ssh-keygen -b 2048 -t rsa -q -N "" -C "" -m pem -f ' . $args{ssh_private_key_file});
}

Reply:
Please check now.